src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -41,20 +42,56 @@ public function __construct(array $sendersMap, ContainerInterface $sendersLocato
      */
     public function getSenders(Envelope $envelope): iterable
     {
-        $seen = [];
+        $senderAliases = [];
 
         foreach (HandlersLocator::listTypes($envelope) as $type) {
+            $typeSenderAliases = [];
+
             foreach ($this->sendersMap[$type] ?? [] as $senderAlias) {
-                if (!\in_array($senderAlias, $seen, true)) {
-                    if (!$this->sendersLocator->has($senderAlias)) {
-                        throw new RuntimeException(sprintf('Invalid senders configuration: sender "%s" is not in the senders locator.', $senderAlias));
-                    }
-
-                    $seen[] = $senderAlias;
-                    $sender = $this->sendersLocator->get($senderAlias);
-                    yield $senderAlias => $sender;
-                }
+                $typeSenderAliases[] = $senderAlias;
+            }
+
+            if (empty($typeSenderAliases)) {
```

Comment:
```suggestion
            if (0 === \count($typeSenderAliases)) {
```

I believe is what we prefer over empty